plan/optimizer.go

```diff
@@ -119,10 +119,10 @@ func doOptimize(flag uint64, logic LogicalPlan, ctx context.Context, allocator *
 	if !AllowCartesianProduct && existsCartesianProduct(logic) {
 		return nil, errors.Trace(ErrCartesianProductUnsupported)
 	}
-	logic.ResolveIndicesAndCorCols()
 	if UseDAGPlanBuilder {
 		return dagPhysicalOptimize(logic)
 	}
+	logic.ResolveIndices()
```

Comment:
dagPhysicalOptimize() may do join reorder, so the column index may be changed. So We  must call  ResolveIndices() in dagPhysicalOptimize().
So we move logic.ResolveIndices() after calling dagPhysicalOptimize(logic)  to avoid doing ResoleIndices work twice.